test/iterable_basics_test.dart

```diff
@@ -407,6 +407,49 @@ void main() {
     });
   });
 
+  group('average', () {
+    test('returns average on list of numbers', () {
+      final nums = <int>[2, 2, 4, 8];
+      final ints = <int>[2, 2, 2];
+      final numbers = [1.5, 1.5, 3, 3];
+
+      expect(nums.average(), 4);
+      expect(ints.average(), 2);
+      expect(ints.average((n) => n * 2), 4);
+      expect(nums.average((n) => n + 1), 5);
+      expect(numbers.average(), 2.25);
+    });
+
+    test('returns 0 on empty lists', () {
+      expect(<int>[].average(), null);
+      expect(<double>[].average(), null);
+      expect(<num>[].average((n) => n * 2), null);
+      expect(<String>[].average((s) => s.length), null);
+      expect(<int>[].average((n) => n * 2), null);
+    });
+
+    test('returns average of custom addend', () {
```

Comment:
nit: returns average of custom `value function`

Reply:
Updated